packages/react-native/ReactCommon/jsc/JSCRuntime.cpp

```diff
@@ -481,6 +482,10 @@ bool JSCRuntime::isInspectable() {
   return false;
 }
 
+void JSCRuntime::setDescription(const std::string &desc) {
```

Comment:
If we really want to support custom descriptions: Since we can't safely call methods on the JSContext here (because threading), and doing it in every call to `evaluateJavaScript` is IMO excessive as mentioned above, I'd replace this with an (optional) constructor parameter. (If we have a need for more constructor parameters in the future, this can evolve into a Hermes-style `RuntimeConfig` object with a builder pattern to cut down on the number of overloads.)

That would, however, require some additional plumbing to pass an argument in from wherever you're constructing your `JSCRuntime`, which I recognise might be a little less convenient than calling a method on the instance you already have.

Reply:
I'll look into the runtime config. I saw that HermesExecutorFactory has a 'setDebuggerName' property so that was my other idea. I'm ok with doing a better long term solution, thanks for helping me point towards one 😃

Reply:
As a general heads up, the exact structure of the debugging setup code in RN will be in a bit of flux over the next few weeks as we work on hoisting some of the CDP implementation from Hermes into RN itself. I expect that we'll make some superficial changes on the JSC side of things to conform with updated interfaces due to that work, and you might run into some merge conflicts if you do anything super elaborate in `JSCExecutorFactory` and friends. But a ` JSCExecutorFactory::setDebuggerName` or similar makes sense to me.

Also, you might want to take a look at the [Android `JSCExecutorFactory`](https://github.com/facebook/react-native/blob/d655d4453293db951980c69e953c4866f3c072fe/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/jscexecutor/JSCExecutorFactory.kt#L14-L26) that seems to already pass in some "app name"-type information (which I'm not sure is being used).

Reply:
I ended up going the runtimeConfig route, and also took heavy inspiration from https://github.com/facebook/react-native/commit/32d12e89f864a106433c8e54c10691d7876333ee on what the API should look like.

It's been... a while since I've written C++ (Mostly an Objective C / Swift developer 🙂), so forgive me for any naive mistakes I may have made. How does this change look to you now?